tests/ui/numbered_fields.rs

```diff
@@ -0,0 +1,34 @@
+//run-rustfix
+#![warn(clippy::numbered_fields)]
+
+#[derive(Default)]
+struct TupleStruct(u32, u32, u8);
+
+// This shouldn't lint because it's in a macro
+macro_rules! tuple_struct_init {
+    () => {
+        TupleStruct { 0: 0, 1: 1, 2: 2 }
+    };
+}
+
+fn main() {
+    let tuple_struct = TupleStruct::default();
+
+    // This should lint
+    let _ = TupleStruct {
+        0: 1u32,
+        1: 42,
+        2: 23u8,
+    };
```

Comment:
Could you also add a test where the fields are out of order? Something like this:

```rs
    let _ = TupleStruct {
        2: 23u8,
        0: 1u32,
        1: 42,
    };
```